opal/corelib/range.rb

```diff
@@ -85,16 +93,26 @@ def exclude_end?
     @exclude
   end
 
-  alias :first :begin
+  def first(n=(n_specified=true; nil))
```

Comment:
Like this?

```
  def first(n=undefined)
    return @begin if `n === undefined`
    super
  end
```

The code I had here was standard Ruby hack (even if it looks ugly), I'm not totally sure when `undefined` can happen in Opal code.